src/cozmo/robot.py

```diff
@@ -1314,11 +1332,15 @@ def start_behavior(self, behavior_type):
         '''
         if not isinstance(behavior_type, behavior._BehaviorType):
             raise TypeError('Invalid behavior supplied')
+
+        if self._current_behavior is not None:
+            self._current_behavior._set_stopped()
+
         b = self.behavior_factory(self, behavior_type, is_active=True, dispatch_parent=self)
         msg = _clad_to_engine_iface.ExecuteBehaviorByExecutableType(
                 behaviorType=behavior_type.id)
         self.conn.send_msg(msg)
-        self._is_behavior_running = True
+        self._current_behavior = b
```

Comment:
I know this is pre-existing but would be good to rename the var from 'b' to 'behavior' or something more explanatory.